### PR TITLE
Clean up repo organization

### PR DESCRIPTION
**Pull request by olix0r**
- interpeter should not require on linkerd-core, only namer-core.

- move ThriftProtocol configs to linkerd-protocol-thrift so that the root config project does not need to pull in thrift dependencies.

- re-order LinkerdBuild.scala so that things are ordered by dependency order: router, namer, namerd, linkerd

### CODE REVIEW

**Comment by siggy** [comment]
:+1: :ship: lgtm

**Comment by adleong** [comment]
I think interpreter pulling in `linkerd-core` is a mistake.  It looks like it can just depend on `namer-core` instead.  I don't think interpreter is linkerd specific and I think it makes sense at the top-level or possibly in `namer`.

File: project/LinkerdBuild.scala [hunk 1]
```diff
@@ -69,7 +69,7 @@ object LinkerdBuild extends Base {
   }
 
   val configCore = projectDir("config")
-    .withTwitterLibs(Deps.finagle("core"), Deps.finagle("thrift"))
+    .withTwitterLibs(Deps.finagle("core"))
```
Comment (adleong) [inline project/LinkerdBuild.scala]: `ThriftProtocolDeserializer` imports `import com.twitter.finagle.thrift.Protocols`  
isn't `finagle-thrift` needed for that?

Comment (gtcampbell) [inline project/LinkerdBuild.scala]: `ThriftProtocolDeserializer` got moved to `linkerd-protocol-thrift`  as part of this PR.

**Comment by olix0r** [comment]
moved interpreter back out of linkerd, changed dependency to Namer.core instead of Linkerd.core

File: project/LinkerdBuild.scala (hunk 1 above)
Comment (adleong) [inline project/LinkerdBuild.scala]: ah, I see now.  ignore me.

**Comment by adleong** [comment]
:star: 🔭 

[Approved with PullApprove](https://pullapprove.com/BuoyantIO/linkerd/pull-request/259/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=BuoyantIO/linkerd)

**Pull request merged**